### ISSUE TITLE
Fatal error if variable name starts with '$--'

### ISSUE THREAD

**Issue by danherd** [sasstools/sass-lint#1183]
**What version of Sass Lint are you using?**

1.12.1

**Please include any relevant parts of your configuration**

.sass-lint.yml in full:

```
rules:
  property-sort-order: 0
  no-color-literals:
    - 1
    -
      allow-rgba: true
```

**What did you do? Please include the actual source code causing the issue.**

`sass-lint ./_element-variables.scss`

_element-variables.scss in full:

```
$--color-primary: #ff0000;
```

**What did you expect to happen?**

There to be no issues with this file.

**What actually happened? Please include any error messages given to you by Sass Lint.**

The following error is shown:

```
/usr/local/lib/node_modules/sass-lint/index.js:313
    throw new exceptions.SassLintFailureError(errorCount.count + ' errors were detected in \n- ' + errorCount.files.join('\n- '));
    ^
SassLintFailureError: 1 errors were detected in
- ./_element-variables.scss
    at Function.sassLint.failOnError (/usr/local/lib/node_modules/sass-lint/index.js:313:11)
    at detectPattern (/usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:31:10)
    at /usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:83:5
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (/usr/local/lib/node_modules/sass-lint/bin/sass-lint.js:82:16)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
```

**If you're using a IDE plugin have you tried the CLI too?**

Yes

**Extra information:**

I am using Element UI for Vue (https://github.com/ElemeFE/element) and this is from their sass variables file, so I can't really change the name formatting myself. Basically, what I want to know is, is this variable name format OK? If so, it needs fixing here and if not, well, I guess I'll need to let the Element guys know!

Cheers.

**Comment by dolcy** [comment]
I'm having the same issue with Element UI.

**Comment by danherd** [comment]
@dolcy I switched to using stylelint (https://stylelint.io) and that doesn't have a problem with the Element UI variable names.

**Comment by dolcy** [comment]
@danherd Thanks...figured that would be the next step; currently using sass-lint with Atom. Cheers.
